source/common/runtime/runtime_features.cc

```diff
@@ -61,6 +61,7 @@ constexpr const char* runtime_features[] = {
     "envoy.reloadable_features.conn_pool_delete_when_idle",
     "envoy.reloadable_features.correct_scheme_and_xfp",
     "envoy.reloadable_features.correctly_validate_alpn",
+    "envoy.reloadable_features.disable_local_interface_name_for_upstream_connection",
```

Comment:
@yanavlasov @alyssawilk I went back and forth on if this should be a worded like this and enabled by default or worded as `enable_*` and then make it part of `disabled_runtime_features`.

Happy to go either way

Reply:
Created https://github.com/envoyproxy/envoy/issues/19759 for a longer term solution.

Reply:
As commented on the issue I think we're going to want to have a permanent config knob rather than the runtime guard.  If you buy that, let's add a TODO here to replace the flag, and make sure you do that wiring in the next month or so?
/wait